src/peft/tuners/boft/layer.py

```diff
@@ -264,6 +264,14 @@ def update_layer(
         """
         Update the linear layer with trainable BOFT weights. Override for other layer types.
         """
+        # Attempt to load the CUDA extension during model initialization
+        if not get_fbd_cuda():
+            self.fbd_cuda_available = False
+            # If the CUDA extension is not available, set the butterfly factor to 1 to speed up the finetuning process
+            boft_n_butterfly_factor = 1
```

Comment:
Should we log this as an info as well?

Reply:
This could be logged, yes. Since this part of the code was not changed in this PR, only moved around, I'll leave that open for a subsequent PR.